src/classes/PMT_PaymentWizard_CTRL.cls

```diff
@@ -374,20 +409,50 @@ public with sharing class PMT_PaymentWizard_CTRL {
         }
     }
 
+    /*******************************************************************************************************
+    * @description Displays an error message when then user has no permission to modify payments
+    * @return void
+    */ 
+    private void displayNoAccessMessage() {
+        String errorMsg;
+
+        if (isWriteoffWizard == true) {
+            errorMsg = System.Label.pmtWizardWriteoffNoAccessMessage;
+        } else if (isScheduleWizard == true) {
+            errorMsg = System.Label.pmtWizardPaymentsNoAccessMessage;
+        } else {
+            errorMsg = System.Label.pmtWizardMsgNoOppFound;
+        }
+
+        ApexPages.addMessage(new ApexPages.Message(ApexPages.Severity.ERROR, errorMsg));
+    }
+
+    /*******************************************************************************************************
+    * @description Returns true if the user has permissions to access, create and delete npe01__oppPayment__c
+    * @return Boolean
+    */ 
+    private Boolean getPaymentAccess() {
+        Schema.DescribeSObjectResult paymentDescribe = npe01__oppPayment__c.sObjectType.getDescribe();
+
+        return paymentDescribe.isCreateable() && paymentDescribe.isDeletable() && paymentDescribe.isAccessible();
```

Comment:
We will not display fields if the user cannot create payments or write them off, so no need to check the field access. If the user has permission to the Payments sObject, but not to the fields, the FLS will be respected, since the page is using `apex:outputField` to display data. I did not test the form if the user has access to the sObject but not specific fields and need to input data (the page is using the `UTIL_FormField`/`UTIL_InputField` which use `apex:inputField` ), but if there is any issue it should be another work item since this one is focused on the page access when the user has no permission to write off payment or create payments. Thoughts?